scripts/cubemx-importer.stanza

```diff
@@ -418,24 +418,43 @@ defn print-i2s (o:OutputStream, mcu:Stm32MCU):
     val pins = unique-pins-for(instance-name(ip-block), mcu)
     val optional = StringBuffer()
 
-    val mck? = contains?(pins, "MCLK")
+    val mck? = contains?(pins, "MCK")
     val sdmi? = contains?(pins, "SDI")
+    val sd? = contains?(pins, "SD")
+    val sdo? = contains?(pins, "SDO")
+    val extsd? = contains?(pins, "ext_SD")
+    val ck? = contains?(pins, "CK")
+    val sck? = contains?(pins, "SCK")
+
     if mck?:
-      print(optional, "I2S-MCK, ")
+      print(optional, "I2S-MCK")
```

Comment:
You need the comma, since `optional` is going to be threaded through to the `i2s([...]`) generator.

Reply:
Comma is at the beginning of the next if statement. It would generate either [I2S-MCK] or [I2S-MCK, I2S-SDMI]. Previously it was generating [I2S-MCK, ] when SDMI was absent

Reply:
I see. We can clean this up a little bit

```
val optional = Vector<String>()

if mck?: 
  add(signals, "I2S-MCK")
```

and use

```
"i2s([%,])"" % [optional]
```
in the printer

Reply:
Done